src/axom/core/Map.hpp

```diff
@@ -0,0 +1,487 @@
+// Copyright (c) 2017-2021, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef AXOM_MAP_HPP_
+#define AXOM_MAP_HPP_
+
+#include "axom/config.hpp"                 // for compile-time defines
+#include "axom/core/Macros.hpp"            // for axom macros
+#include "axom/core/memory_management.hpp" // for memory allocation functions
+#include "axom/core/Types.hpp"             // for axom types
+
+
+// C/C++ includes
+#include <functional> //for hashing until a custom method is defined
+#include <iostream>
+
+namespace axom
+{
+
```

Comment:
I wonder if this should be in an "experimental" namespace  until we've settled on an API that we want to expose to users.  E.g. `axom::experimental` ? (or something better)

Thoughts?

Reply:
I personally think this is a great idea, but I'm uncertain if that's my decision to make -- what's the best way to have this conversation?

Reply:
Me have just had it :)

Any objections: @rhornung67 (and/or others) to placing `axom::Map` in `axom::experimental::Map` as a way of getting the code in early, without worrying about a finalized API at this stage?

Reply:
Sounds good.

Reply:
After @ksuarez1423 pointed me at this: this namespace should have the redundant `axom_` removed from it... so:

`axom::experimental::axom_map` to `axom::experimental::map`